src/uucore/src/lib/features/uptime.rs

```diff
@@ -0,0 +1,283 @@
+// spell-checker:ignore gettime BOOTTIME clockid boottime formated nusers loadavg getloadavg
+
+use crate::error::{UError, UResult};
+use chrono::Local;
+use libc::time_t;
+use thiserror::Error;
+
+#[derive(Debug, Error)]
+pub enum UptimeError {
+    #[error("could not retrieve system uptime")]
+    SystemUptime,
+    #[error("could not retrieve system load average")]
+    SystemLoadavg,
+    #[error("boot time larger than current time")]
+    BootTime,
+}
+
+impl UError for UptimeError {
+    fn code(&self) -> i32 {
+        1
+    }
+}
+
+pub fn get_formatted_time() -> String {
+    Local::now().time().format("%H:%M:%S").to_string()
+}
+
+#[cfg(target_os = "openbsd")]
+pub fn get_uptime(_boot_time: Option<time_t>) -> UResult<i64> {
```

Comment:
given that this functions are going to be exposed and proposed https://docs.rs/uucore/latest/uucore/
it would be nice to have some rustdoc comments for these pub functions :)